x/ccv/provider/keeper/grpc_query_test.go

```diff
@@ -54,3 +55,36 @@ func TestQueryAllPairsValConAddrByConsumerChainID(t *testing.T) {
 	require.Equal(t, &consumerKey, response.PairValConAddr[0].ConsumerKey)
 	require.Equal(t, &expectedResult, response.PairValConAddr[0])
 }
+
+func TestQueryOldestUnconfirmVsc(t *testing.T) {
+	chainID := consumer
+
+	pk, ctx, ctrl, _ := testkeeper.GetProviderKeeperAndCtx(t, testkeeper.NewInMemKeeperParams(t))
+	defer ctrl.Finish()
+
+	var vscID uint64 = 1
+	now := time.Now().UTC()
+	pk.SetVscSendTimestamp(ctx, chainID, vscID, now)
+	pk.SetConsumerClientId(ctx, chainID, "client-1")
+
+	// Request is nil
+	_, err := pk.QueryOldestUnconfirmVsc(ctx, nil)
+	require.Error(t, err)
+
+	// Request with chainId is empty
+	_, err = pk.QueryOldestUnconfirmVsc(ctx, &types.QueryOldestUnconfirmVscRequest{})
+	require.Error(t, err)
+
+	// Request with chainId is invalid
+	_, err = pk.QueryOldestUnconfirmVsc(ctx, &types.QueryOldestUnconfirmVscRequest{ChainId: "invalidChainId"})
+	require.Error(t, err)
+
+	// Request is valid
```

Comment:
Might make sense to test with at least 2 `VscSendTimestamp`s here to verify that the oldest is returned (?)